src/main/drivers/timer_stm32f10x.c

```diff
@@ -23,25 +23,56 @@
 #include "rcc.h"
 #include "timer.h"
 
-const timerDef_t timerDefinitions[HARDWARE_TIMER_DEFINITION_COUNT] = {
+#define TIM_N(n) (1 << (n))
+#define USED_TIMER_COUNT BITCOUNT(USED_TIMERS)
+
+const timerDef_t timerDefinitions[USED_TIMER_COUNT] = {
+#if USED_TIMERS & TIM_N(1)
     { .TIMx = TIM1,  .rcc = RCC_APB2(TIM1), .inputIrq = TIM1_CC_IRQn },
+#endif
+#if USED_TIMERS & TIM_N(2)
     { .TIMx = TIM2,  .rcc = RCC_APB1(TIM2), .inputIrq = TIM2_IRQn },
+#endif
+#if USED_TIMERS & TIM_N(3)
     { .TIMx = TIM3,  .rcc = RCC_APB1(TIM3), .inputIrq = TIM3_IRQn },
+#endif
+#if USED_TIMERS & TIM_N(4)
     { .TIMx = TIM4,  .rcc = RCC_APB1(TIM4), .inputIrq = TIM4_IRQn },
+#endif
 #if defined(STM32F10X_HD) || defined(STM32F10X_CL) || defined(STM32F10X_XL) || defined(STM32F10X_HD_VL)
```

Comment:
This is a bit dangerous - there must be `timerDefinitions` entry for every USED_TIMERS bit. If someone enables TIM_N(5) on target that does not support it, following entries will shift and it will lead to nasty bug.

One possibility is to define dummy entry in that case.
Another possibility is to trigger error - IMO it is better approach - force target definitions to enable only timers that are available on target MCU:
```
#if USED_TIMERS & TIM_N(5)
 #if defined(STM32F10X_HD) || defined(STM32F10X_CL) || defined(STM32F10X_XL) || defined(STM32F10X_HD_VL)
    { .TIMx = TIM5,  .rcc = RCC_APB1(TIM5), .inputIrq = TIM5_IRQn },
 #else
   #error "TIM5 is not supported on this CPU"
 #endif
#endif